pkg/operator/options/options.go

```diff
@@ -57,23 +83,113 @@ func New() *Options {
+	f.StringVar(&opts.featureGates, "feature-gates", env.WithDefaultString("FEATURE_GATE", ""), "TODO")
+
 	if opts.MemoryLimit > 0 {
 		newLimit := int64(float64(opts.MemoryLimit) * 0.9)
 		debug.SetMemoryLimit(newLimit)
 	}
+
 	return opts
 }
 
-// MustParse reads the user passed flags, environment variables, and default values.
-// Options are valided and panics if an error is returned
-func (o *Options) MustParse() *Options {
-	err := o.Parse(os.Args[1:])
-
-	if errors.Is(err, flag.ErrHelp) {
+func (*Options) Inject(ctx context.Context) (context.Context, error) {
+	opts := New()
+	if err := opts.Parse(os.Args[1:]); errors.Is(err, flag.ErrHelp) {
 		os.Exit(0)
+	} else if err != nil {
+		return ctx, err
+	}
+
+	// Check if BatchMaxDuration and BatchIdleDuration have been set. If they have not been set, they will be overwritten by
+	// values in the configmap, if present.
+	opts.Visit(func(f *flag.Flag) {
+		switch f.Name {
+		case "batch-max-duration":
+			opts.BatchMaxDurationSet = true
+		case "batch-idle-duration":
+			opts.BatchIdleDurationSet = true
+		}
+	})
+	if _, ok := os.LookupEnv("BATCH_MAX_DURATION"); ok {
+		opts.BatchMaxDurationSet = true
+	}
+	if _, ok := os.LookupEnv("BATCH_IDLE_DURATION"); ok {
+		opts.BatchIdleDurationSet = true
 	}
-	if err != nil {
-		panic(err)
+
+	if opts.featureGates != "" {
+		gates := parseFeatureGates(opts.featureGates)
+		val, ok := gates["Drift"]
+		opts.DriftEnabled = lo.Ternary(ok, val, false)
+		opts.DriftEnabledSet = ok
+	}
+
+	return ToContext(ctx, opts), nil
+}
+
+// MergeSettings applies settings specified in the v1alpha5 configmap to options. If the value was already specified by
+// a CLI argument or environment variable, that value will be used.
+func (*Options) MergeSettings(ctx context.Context, injectables ...settings.Injectable) {
+	opts := FromContext(ctx)
+	injectables = lo.Map(injectables, func(in settings.Injectable, _ int) settings.Injectable {
+		return in.FromContext(ctx)
+	})
+
+	for _, injectable := range injectables {
+		setting, ok := injectable.(*settings.Settings)
+		if !ok {
+			continue
+		}
+
+		if !opts.BatchMaxDurationSet {
+			opts.BatchMaxDuration = setting.BatchMaxDuration
+			opts.BatchMaxDurationSet = true
+		}
+		if !opts.BatchIdleDurationSet {
+			opts.BatchIdleDuration = setting.BatchIdleDuration
+			opts.BatchIdleDurationSet = true
+		}
+		if !opts.DriftEnabledSet {
+			opts.DriftEnabled = setting.DriftEnabled
+			opts.DriftEnabledSet = true
+		}
+	}
+}
+
+func ToContext(ctx context.Context, opts *Options) context.Context {
+	return context.WithValue(ctx, optionsKey{}, opts)
+}
+
+func FromContext(ctx context.Context) *Options {
+	retval := ctx.Value(optionsKey{})
+	if retval == nil {
+		// This is a developer error if this happens, so we should panic
+		panic("options doesn't exist in context")
+	}
+	return retval.(*Options)
+}
+
+func parseFeatureGates(val string) map[string]bool {
```

Comment:
How does the upstream apiserver parse these args? Is it the same method? Can we leverage their parsing logic?